dms/models/directory.py

```diff
@@ -638,7 +647,11 @@ def create(self, vals_list):
                 parent = self.browse([vals["parent_id"]])
                 data = next(iter(parent.sudo().read(["storage_id"])), {})
                 vals["storage_id"] = self._convert_to_write(data).get("storage_id")
-        return super().create(vals_list)
+        # Hack to prevent error related to mail_message parent not exists in some cases
+        ctx = dict(self.env.context).copy()
+        ctx.update({"default_parent_id": False})
+        res = super(DmsDirectory, self.with_context(ctx)).create(vals_list)
```

Comment:
This code is easy to simplify:

```suggestion
        # Hack to prevent error related to mail_message parent not exists in some cases
        res = super(DmsDirectory, self.with_context(default_parent_id=False)).create(vals_list)
```

I see that you were able to remove this HACK due to the new v13 ORM: https://github.com/OCA/dms/blob/db9cb933ba4f29cc1a610190c32e2ff4bb16b1ce/dms/models/directory.py#L654-L660

Nice!